icevision/visualize/draw_data.py

```diff
@@ -77,36 +150,51 @@ def draw_label(
         caption = class_map.get_id(label)
     else:
         caption = str(label)
-
-    return _draw_label(img=img, caption=caption, x=int(x), y=int(y), color=color)
+    if prettify:
+        # We could introduce a callback here for more complex label renaming
+        caption = prettify_func(caption)
+
+    # Append label confidence to caption if applicable
+    if score is not None:
+        if prettify:
+            score = f"{score * 100: .2f}%"
+        caption = f"{caption}: {score}"
+
+    if not Path(font).exists():
+        # PIL throws cryptic errors for wrong filepaths, so let's catch it earlier here
+        raise FileNotFoundError(f"{font} file doesn't exist")
+    return _draw_label(
+        img=img,
+        caption=caption,
+        x=x,
+        y=y,
+        color=color,
+        font_path=font,
+        font_size=int(font_size),
+        return_as_pil_img=return_as_pil_img,
+    )
 
 
 def _draw_label(
     img: np.ndarray,
     caption: str,
     x: int,
     y: int,
-    color,
-    font=cv2.FONT_HERSHEY_SIMPLEX,
-    font_scale: float = 1.0,
-):
-    """Draws a caption above the box in an image."""
-    y -= 10
-    w, h = cv2.getTextSize(caption, font, fontScale=font_scale, thickness=1)[0]
-
-    # make the coords of the box with a small padding of two pixels
-    # check if the box_pt2 is inside the image otherwise invert the label box (meaning the label box will be inside the bounding box)
-    if (y - h - 2) > 0:
-        box_pt1, box_pt2 = (x, y + 10), (x + w + 2, y - h - 2)
+    color: Union[np.ndarray, list, tuple],
+    font_path=DEFAULT_FONT_PATH,
+    font_size: int = 20,
+    return_as_pil_img: bool = False,
+) -> Union[PIL.Image.Image, np.ndarray]:
+    """Draw labels on the image"""
+    font = PIL.ImageFont.truetype(font_path, size=font_size)
+    xy = (x + 10, y + 5)
+    img = PIL.Image.fromarray(img)
+    draw = ImageDraw.Draw(img)
+    draw.text(xy, caption, font=font, fill=as_rgb_tuple(color))
+    if return_as_pil_img:
+        return img
     else:
-        box_pt1, box_pt2 = (x, y + h + 22), (x + w + 2, y + 10)
-
-    cv2.rectangle(img, box_pt1, box_pt2, color, cv2.FILLED)
-
-    label_pt = (box_pt1[0], box_pt1[1] - 10)
-    cv2.putText(img, caption, label_pt, font, font_scale, (240, 240, 240), 2)
-
-    return img
+        return np.array(img)
 
 
 def draw_record(
```

Comment:
Lotta code smell here. In a future PR, maybe we could:
* Directly write the addition `record.load()` in `show_record` itself so we don't need this function call
* Move to `**kwargs`?

Reply:
Same thing applies to `draw_pred` as well